tests/unit/logic/vote.spec.js

```diff
@@ -0,0 +1,1012 @@
+				sender = {address: 123}
+				Vote.__set__('setImmediate', setImmediate)
+
+				vote.bind(delegates, rounds, system)
+			})
+
+			afterEach(function () {
+				clock.restore()
+			})
+
+			context('if trs.asset.votes is null', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: null}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is empty', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: []}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is undefined', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: undefined}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes has not a valid format', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: [1,2]}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should not call to scope.account.merge()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.false
+					expect(callback.called).to.be.true
+				})
+			})
+
+			context('if trs.asset.votes is an Array and have items', function () {
+
+				beforeEach(function () {
+					trs = {asset: {votes: ['+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22', '+b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9']}}
+					vote.undoUnconfirmed(trs, sender, callback)
+				})
+
+				it('should call to scope.account.merge() and callback()', function () {
+					clock.runAll()
+					var result = Vote.__get__('self.scope')
+					expect(result.account.merge.called).to.be.true
+					expect(callback.called).to.be.true
+				})
+			})
+		})
+
+		describe('objectNormalize()', function () {
+      
+			var trs_success, trs_fail_1, trs_fail_2, trs_fail_3, trs_fail_4, trs_fail_5, trs_fail_6, trs_fail_7, trs_fail_8, vote, logger, schema, execMethod
+      
+			beforeEach(function () {
+				trs_success = {asset: {votes: ['+37b6d3fc89fb418fa5e3799e760cc2b1733a567daebb08a531d3d7e8b24f2d22', '-6d1103674f29502c873de14e48e9e432ec6cf6db76272c7b0dad186bb92c9a9a']}}
+				trs_fail_1 = {asset: undefined}
```

Comment:
There is no need to assign all this variables for each test. I would move this logic inside each test.

E.g. trs_fail_5 is only good for the test #5, so it's useless code in the test #1...

Reply:
Done!